### DISCUSSION TITLE
How can I fix this, Type int_const 7781...(69 digits omitted)...9408 is not implicitly convertible to expected type int256. Literal is too large to fit in int256.

### DISCUSSION THREAD

**Discussion by YavorJJ** [Cyfrin/foundry-full-course-cu#1954] [Q&A] [answered]
Hello I recently got into a problem with the anvil private key, in the int256. 

![image](https://github.com/Cyfrin/foundry-full-course-cu/assets/171678195/e43dbebb-a6c8-4795-b584-7dd1516fa9a1)

Here is my code:
' ' ' solidity

 // SPDX-License-Identifier: MIT

pragma solidity ^0.8.19;

import {Script} from "forge-std/Script.sol";
import {MockV3Aggregator} from "../test/mocks/MockV3Aggregator.sol";
import {ERC20Mock} from "@openzeppelin/contracts/mocks/token/ERC20Mock.sol";

contract HelperConfig is Script {
    struct NetworkConfig {
        address wethUsdPriceFeed;
        address wbtcUsdPriceFeed;
        address weth;
        address wbtc;
        uint256 deployerKey;
    }
    
    uint8 public constant DECIMALS = 8;
    int256 public constant ETH_USD_PRICE = 2000e8;
    int256 public constant BTC_USD_PRICE = 1000e8;
    int256 public DEFAULT_ANVIL_KEY = 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80;

    NetworkConfig public activeNetworkconfig;

    constructor() {} 

    function getSepoliaEthConfig() public view returns (NetworkConfig memory) {
        return NetworkConfig({
            wethUsdPriceFeed: 0x694AA1769357215DE4FAC081bf1f309aDC325306, // ETH / USD
            wbtcUsdPriceFeed: 0x1b44F3514812d835EB1BDB0acB33d3fA3351Ee43,
            weth: 0xdd13E55209Fd76AfE204dBda4007C227904f0a81,
            wbtc: 0x8f3Cf7ad23Cd3CaDbD9735AFf958023239c6A063,
            deployerKey: vm.envUint("PRIVATE_KEY")
        });
    }

    function getOrCreateAnvilEthConfig() public returns(NetworkConfig memory) {
        if(activeNetworkconfig.wethUsdPriceFeed != address(0)){
            return activeNetworkconfig;
        }

        vm.startBroadcast();
        MockV3Aggregator ethUsdPriceFeed = new MockV3Aggregator(
            DECIMALS, 
            ETH_USD_PRICE
        );
        ERC20Mock wethMock = new ERC20Mock("WETH", "WETH", msg.sender, 1000e8);

         vm.startBroadcast();
        MockV3Aggregator btcUsdPriceFeed = new MockV3Aggregator(
            DECIMALS, 
            BTC_USD_PRICE
        );
        ERC20Mock btcMock = new ERC20Mock("WBTC", "WBTC", msg.sender, 1000e8);
        vm.stopBroadcast();

        return NetworkConfig({
            wethUsdPriceFeed: address(ethUsdPriceFeed), 
            wbtcUsdPriceFeed: address(btcUsdPriceFeed),
            weth: address(wethMock),
            wbtc: address(btcMock),
            deployerKey: vm.envUint("DEFAULT_ANVIL_KEY")
        });
    }

}

' ' '

every help is appreciated!

**Comment by EngrPips** [comment, marked as answer]
The below is my `HelperConfig` file that works

```js
// SPDX-License-Identifier: UNLICENSED
pragma solidity 0.8.23;

import {Script} from "forge-std/Script.sol";
import {ERC20Mock} from "@openzeppelin/contracts/mocks/ERC20Mock.sol";
import {MockV3Aggregator} from "../test/mocks/MockV3Aggregator.sol";

contract HelperConfig is Script {
    struct Network_Config {
        address wETH_token_address;
        address wBTC_token_address;
        address wETH_token_addres_price_feed;
        address wBTC_token_addres_price_feed;
        uint256 deployer_key;
    }

    // WETH TOKEN DETAILS
    string private constant WETH_NAME = "WETH";
    string private constant WETH_SYMBOL = "WETH";
    uint256 private constant WETH_INITIAL_BALANCE = 100e18;
    uint8 private constant WETH_TOKEN_DECIMALS = 8;
    int256 private constant WETH_TOKEN_INITIAL_PRICE = 1000e8;

    // WBTC TOKEN DETAILS
    string private constant WBTC_NAME = "WBTC";
    string private constant WBTC_SYMBOL = "WBTC";
    uint256 private constant WBTC_INITIAL_BALANCE = 100e18;
    uint8 private constant WBTC_TOKEN_DECIMALS = 8;
    int256 private constant WBTC_TOKEN_INITIAL_PRICE = 2000e8;

    // ANVIL DETAILS
    uint256 private constant DEFAULT_ANVIL_KEY = 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80;

    Network_Config public active_network_config;

    constructor() {
        if (block.chainid == 11155111) active_network_config = get_sepolia_network_config();
        if (block.chainid == 31337) active_network_config = get_anvil_network_config();
    }

    function get_sepolia_network_config() public view returns (Network_Config memory sepolia_network_config) {
        sepolia_network_config = Network_Config({
            wETH_token_address: 0xdd13E55209Fd76AfE204dBda4007C227904f0a81,
            wBTC_token_address: 0x8f3Cf7ad23Cd3CaDbD9735AFf958023239c6A063,
            wETH_token_addres_price_feed: 0x694AA1769357215DE4FAC081bf1f309aDC325306,
            wBTC_token_addres_price_feed: 0x1b44F3514812d835EB1BDB0acB33d3fA3351Ee43,
            deployer_key: vm.envUint("SEPOLIA_PRIVATE_KEY_ADDRESS")
        });

        return sepolia_network_config;
    }

    function get_anvil_network_config() public returns (Network_Config memory anvil_network_config) {
        vm.startBroadcast();
        ERC20Mock wETH = new ERC20Mock(WETH_NAME, WETH_SYMBOL, msg.sender, WETH_INITIAL_BALANCE);
        MockV3Aggregator wETH_price_feed = new MockV3Aggregator(WETH_TOKEN_DECIMALS, WETH_TOKEN_INITIAL_PRICE);

        ERC20Mock wBTC = new ERC20Mock(WBTC_NAME, WBTC_SYMBOL, msg.sender, WBTC_INITIAL_BALANCE);
        MockV3Aggregator wBTC_price_feed = new MockV3Aggregator(WBTC_TOKEN_DECIMALS, WBTC_TOKEN_INITIAL_PRICE);
        vm.stopBroadcast();

        anvil_network_config = Network_Config({
            wETH_token_address: address(wETH),
            wBTC_token_address: address(wBTC),
            wETH_token_addres_price_feed: address(wETH_price_feed),
            wBTC_token_addres_price_feed: address(wBTC_price_feed),
            deployer_key: DEFAULT_ANVIL_KEY
        });

        return anvil_network_config;
    }
}

```
Maybe try using uint256 instead of int256 and see if it will work.

**Reply by YavorJJ** [reply]
yep, its uint256. thank you.

**Reply by EngrPips** [reply]
Awesome, I am glad you figured it out.
